src/xdp/program.c

```diff
@@ -1387,6 +1562,112 @@ XdpProgramTraceObject(
     }
 }
 
+static const ebpf_context_descriptor_t EbpfXdpContextDescriptor = {
+    .size = sizeof(xdp_md_t),
+    .data = FIELD_OFFSET(xdp_md_t, data),
+    .end = FIELD_OFFSET(xdp_md_t, data_end),
+    .meta = FIELD_OFFSET(xdp_md_t, data_meta),
+};
+
+#define XDP_EXT_HELPER_FUNCTION_START EBPF_MAX_GENERAL_HELPER_FUNCTION
+
+// XDP helper function prototype descriptors.
+static const ebpf_helper_function_prototype_t EbpfXdpHelperFunctionPrototype[] = {
+    {
+        .helper_id = XDP_EXT_HELPER_FUNCTION_START + 1,
+        .name = "bpf_xdp_adjust_head",
+        .return_type = EBPF_RETURN_TYPE_INTEGER,
+        .arguments = {
+            EBPF_ARGUMENT_TYPE_PTR_TO_CTX,
+            EBPF_ARGUMENT_TYPE_ANYTHING,
+        },
+    },
+};
+
+// TODO: fix and upstream const-correct definitions
```

Comment:
Track/comment with issue?

Reply:
Yup filed #169 in our repo and https://github.com/microsoft/ebpf-for-windows/issues/2145